app/models/host/managed.rb

```diff
@@ -673,8 +689,13 @@ def root_pass
   end
 
   def clone
+    aspects = host_aspects.map do |aspect|
+      assoc = Host.reflect_on_all_associations.find {|a| a.class_name == aspect.execution_model_type}
+      assoc.name.to_sym
+    end
+
     # do not copy system specific attributes
-    host = self.deep_clone(:include => [:host_config_groups, :host_classes, :host_parameters],
+    host = self.deep_clone(:include => [:host_config_groups, :host_classes, :host_parameters, :host_aspects] + aspects,
```

Comment:
That is a tad ugly, but WFM!

On Sun, Sep 6, 2015 at 9:19 AM, Shimon Shtein <notifications@github.com>
wrote:

> In app/models/host/managed.rb
> <https://github.com/theforeman/foreman/pull/2665#discussion_r38821355>:
>
> >      # do not copy system specific attributes
> > -    host = self.deep_clone(:include => [:host_config_groups, :host_classes, :host_parameters],
> > +    host = self.deep_clone(:include => [:host_config_groups, :host_classes, :host_parameters, :host_aspects] + aspects,
>
> Theoretically it is implemented using the #after_clone hook. You can
> always remove anything from the object after it was cloned.
> I agree this is not the best thing, but I would prefer a slimmer interface
> on the aspect model.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/theforeman/foreman/pull/2665/files#r38821355>.
>